### PR TITLE
[14.0][IMP] stock_move_auto_assign_auto_release: Release release_ready moves together

### PR DESCRIPTION
**Pull request by mt-software-de**
A a transfer is waiting for more than product to release and the release policy is "one".
The whole transfer should be released, instead of only the last received one. 

Within this PR
Also releaseable siblings will be released.

### CODE REVIEW

**Comment by mt-software-de** [comment]
cc @lmignon @TDu @jbaudoux

**Review by lmignon — COMMENTED** [review]
LGTM (Code review only)

**Review by jbaudoux — COMMENTED** [review]
What about dropping this `moves_auto_release` method and in `_enqueue_auto_assign`, check that the `move.picking_id` is `is_auto_release_allowed`, and then use as delayable `_delay_auto_release_available_to_promise` (which is better placed in stock_move_auto_assign_auto_release than in stock_release_channel_auto_release)

**Comment by jbaudoux** [comment]
> A a transfer is waiting for more than product to release and the release policy is "one". The whole transfer should be released, instead of only the last received one.

It would be worth adding a fix on `stock_available_to_promise_release` so that when `release_available_to_promise` is called on `stock.move` it raises an error. And on `stock.picking` `release_available_to_promise`, call `moves...._run_stock_rule` instead of `moves....release_available_to_promise`

**Comment by mt-software-de** [comment]
> `stock_available_to_promise_release`

**Comment by mt-software-de** [comment]
> > A a transfer is waiting for more than product to release and the release policy is "one". The whole transfer should be released, instead of only the last received one.
> 
> It would be worth adding a fix on `stock_available_to_promise_release` so that when `release_available_to_promise` is called on `stock.move` it raises an error. And on `stock.picking` `release_available_to_promise`, call `moves...._run_stock_rule` instead of `moves....release_available_to_promise`

So you mean a check if all moves of a picking are in the recordset if the policy is one?

**Comment by mt-software-de** [comment]
> What about dropping this `moves_auto_release` method and in `_enqueue_auto_assign`, check that the `move.picking_id` is `is_auto_release_allowed`, and then use as delayable `_delay_auto_release_available_to_promise` (which is better placed in stock_move_auto_assign_auto_release than in stock_release_channel_auto_release)

Ok you mean moving `_delay_auto_release_available_to_promise` into stock_move_auto_assign_auto_release - yes this sounds good.
Needs to be merged before https://github.com/OCA/wms/pull/566

File: stock_move_auto_assign_auto_release/tests/test_assign_auto_release.py [hunk 1]
```diff
@@ -166,3 +173,48 @@ def test_move_field_is_auto_release_allowed(self):
...
+            self.assertFalse(job)
+        with trap_jobs() as trap:
+            self._receive_product(self.product2, 100)
+            shipping.invalidate_cache()
+            shipping.move_lines.invalidate_cache()
+            trap.perform_enqueued_jobs()
+            job = self._get_job_for_method(
+                trap.enqueued_jobs, shipping.auto_release_available_to_promise
+            )
+            job.perform()
```
Comment (jbaudoux) [inline stock_move_auto_assign_auto_release/tests/test_assign_auto_release.py:218]: This is failing
https://github.com/OCA/stock-logistics-warehouse/actions/runs/10195951314/job/28205684041?pr=2122#step:8:1118

Comment (mt-software-de) [reply stock_move_auto_assign_auto_release/tests/test_assign_auto_release.py:218]: I don't know why yet, i will have a look.

Comment (mt-software-de) [reply stock_move_auto_assign_auto_release/tests/test_assign_auto_release.py:218]: Locally it works. :-)

Comment (mt-software-de) [reply stock_move_auto_assign_auto_release/tests/test_assign_auto_release.py:218]: Somehow the trap jobs wasn't working as expected with the current queue.job version. I fixed it

**Review by jbaudoux — APPROVED** [review]

File: stock_move_auto_assign_auto_release/models/product_product.py [hunk 2]
```diff
@@ -14,16 +15,15 @@ def _moves_auto_release_domain(self):
...
 
-    def moves_auto_release(self):
-        """Job trying to auto release moves based on product
+    def pickings_auto_release(self):
+        """Job trying to auto release pickings based on product
 
-        It searches all* the moves auto releasable and trigger the release
-        available to promise process.
+        It searches all* the moves auto releasable
+        and trigger a delayed release available to promise for there pickings.
```
Comment (TDu) [inline stock_move_auto_assign_auto_release/models/product_product.py]:
```suggestion
        and trigger a delayed release available to promise for their pickings.
```

**Review by TDu — APPROVED** [review]
Maybe the readme could be updated also ?
LG otherwise.

File: stock_move_auto_assign_auto_release/models/product_product.py (hunk 2 above)
Comment (jbaudoux) [reply stock_move_auto_assign_auto_release/models/product_product.py]: triggers

**Pull request merged**